packages/babel-plugin-transform-numeric-separator/README.md

```diff
@@ -0,0 +1,128 @@
+# babel-plugin-transform-numeric-separator
+
+> This plugin allows Babel to transform Decimal, Binary, Hex and Octal literals containing Numeric Literal Separator to their non-separated form.
+
+## Example
+
+### Decimal Literals
+
+```js
+let trillion = 1_000_000_000_000;
+console.log(trillion); // 1000000000000
+console.log(trillion === 10 ** 12); // true
+```
+
+### Binary Literals
+
+```js
+let nibbles = 0b1010_0001_1000_0101;
+console.log(nibbles); // 1000
+
+// Is bit 7 on? It sure is!
+// 0b1010_0001_1000_0101
+//             ^
+//             
+// We can double check: 
+console.log(!!(nibbles & (1 << 7))); // true
+```
+
+### Hex Literal
+
+```js
+// Messages are sent as 24 bit values, but should be 
+// treated as 3 distinct bytes:
+let message = 0xA0_B0_C0;
+
+// What's the value of the upper most byte? It's A0, or 160.
+// We can confirm that:
+let a = (message >> 16) & 0xFF; 
+console.log(a.toString(16), a); // a0, 160
+
+// What's the value of the middle byte? It's B0, or 176.
+// Let's just make sure...
+let b = (message >> 8) & 0xFF;
+console.log(b.toString(16), b); // b0, 176
+
+// What's the value of the lower most byte? It's C0, or 192.
+// Again, let's prove that:
+let c = message & 0xFF;
+console.log(c.toString(16), b); // c0, 192
+```
+
+### Octal Literal
+
+*hand wave emoji*
+
+Octals are great for permissions, but also look better when represented in `0o0000` form. No real benefit with separators here. 
+
+## Installation
+
+```sh
+npm install --save-dev babel-plugin-transform-numeric-separator
+```
+
+## Usage
+
+### Via `.babelrc` (Recommended)
+
+**.babelrc**
+
+```json
+{
+  "plugins": ["transform-numeric-separator"]
+}
+```
+
+### Via CLI
+
+```sh
+babel --plugins transform-numeric-separator script.js
+```
+
+### Via Node API
+
+```javascript
+require("babel-core").transform("code", {
+  plugins: ["transform-numeric-separator"]
+});
+```
+
+## Options
+
+### `toDecimal`
+
+`boolean`, defaults to `false`.
+
+By default, this plugin will only remove the separators, printing Decimal, Binary, Hex and Octal number representations. Enabling this option will  convert them all to a Decimal value.
```

Comment:
extra space here?

We do have a http://babeljs.io/docs/plugins/transform-es2015-literals/ to deal with this already so in the ideal case this plugin would defer to that for the transformation. Not sure how that fits but interesting in relation to https://github.com/babel/babel/pull/5735. I'm ok with doing this atm though.